tests/test.rs

```diff
@@ -19,6 +25,42 @@ lazy_static! {
     static ref FILTER_ALL: Regex = Regex::new(".*").unwrap();
 }
 
+fn gen_image_name() -> PathBuf {
+    let mut path = PathBuf::new();
+    path.push("/tmp");
+
+    let id = rand::thread_rng().gen_range(100_000..1_000_000);
+    let image = format!("/tmp/image-{id}.qcow2");
+    path.push(image);
+
+    path
+}
+
+fn create_new_image(image: PathBuf) -> Option<PathBuf> {
+    let out_image = gen_image_name();
+    let out = Command::new("qemu-img")
+        .arg("create")
+        .arg("-F")
+        .arg("raw")
+        .arg("-b")
+        .arg(image)
+        .arg("-f")
+        .arg("qcow2")
+        .arg(out_image.clone())
+        .output()
+        .expect("error creating image file");
+    if !out.status.success() {
+        error!(
+            "error creating image file: out={} err={} status={}",
+            std::str::from_utf8(&out.stdout).unwrap(),
+            std::str::from_utf8(&out.stderr).unwrap(),
+            out.status
+        );
+        return None;
+    }
+    Some(out_image)
```

Comment:
Would be good to have some way to automate cleanup. Otherwise multiple test runs can mess up tmpfs pretty bad given images can be large

Reply:
True. the images end up being ~5 MiB max in my tests, but I should probably clean them up. Note that we don't clean up sockets and other things. Also this doesn't quite pass yet, still iterating...

I suspect the tests are failing due to:

1) We use dirty images for each test (ie, actions take in previous tests can affect actions in subsequent tests even if we take care to not modify the filesystem, UEFI will still modify the EFI partition).
2) UEFI in qemu doesn't seem to have the default EFI binary in the efi vars as a boot option after the first boot and I don't understand why...

Reply:
Ok np. I filed #25. I'll pick it up tmr morning if I wake up early enough